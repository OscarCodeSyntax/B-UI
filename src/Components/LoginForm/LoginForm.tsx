import React from "react";
import { useForm } from "react-hook-form";
import "./../../App.css";
import { FormControl, FormLabel, TextField } from "@mui/material";
 
function LoginForm() {
    const {
        register,
        handleSubmit,
        formState: { errors },
    } = useForm();
 
    const onSubmit = (data:any) => {
        const userData = JSON.parse(localStorage.getItem(data.email));
        if (userData) { // getItem can return actual value or null
            if (userData.password === data.password) {
                console.log(userData.name + " You Are Successfully Logged In");
            } else {
                console.log("Email or Password is not matching with our record");
            }
        } else {
            console.log("Email or Password is not matching with our record");
        }
    };
    
    return (
        <div className="padding-20">
            <h1>Login</h1>
            <form className="App" onSubmit={handleSubmit(onSubmit)}>
            <FormControl className="padding-10">
            <FormLabel>Email</FormLabel>
                <TextField type="email" {...register("email", { required: true })} name='username'/>
                {errors.email && <span style={{ color: "red" }}>
                    *Email* is mandatory </span>}
                    <FormLabel>Password</FormLabel>
                <TextField type="password" {...register("password")} name='password'/>
                <TextField type={"submit"}  defaultValue="Outlined" color="secondary"  value="submit"  name="submit"/>
                </FormControl>
            </form>
        </div>
    );
}
export default LoginForm;