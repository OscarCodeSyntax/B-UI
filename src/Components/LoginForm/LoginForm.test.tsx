import React from "react";
import LoginForm from "./LoginForm";

import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";

describe("LoginForm  ", () => {
  it("renders correctly", () => {
    const ApiButtonsWrapper = renderer.create(<LoginForm />).toJSON();
    expect(ApiButtonsWrapper).toMatchSnapshot();
  });

  it("renders all expected elements", () => {
    const testRenderer = renderer.create(<LoginForm />);
    const testInstance = testRenderer.root;

    expect(testInstance.findAllByType("input").length).toEqual(3);
    expect(testInstance.findAllByType("label").length).toEqual(3);
    expect(testInstance.findAllByType("h1").length).toEqual(1);
  });

  it("has all the correct text", () => {
    render(<LoginForm />);
    expect(screen.getByText("Login")).toBeTruthy();
    expect(screen.getByText("Username")).toBeTruthy();
    expect(screen.getByText("Password")).toBeTruthy();
    expect(screen.getByLabelText("Submit button")).toBeTruthy();
  });
});
