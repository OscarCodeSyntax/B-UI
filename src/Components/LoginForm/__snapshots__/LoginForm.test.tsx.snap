// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm   renders correctly 1`] = `
<div
  className="padding-20"
>
  <h1>
    Login
  </h1>
  <form
    className="App"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root padding-10 css-1nrlq1o-MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
      >
        Username
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            name="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="username"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-ihdtdm"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <label
        className="MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root"
      >
        Password
      </label>
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r1:"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-ihdtdm"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <br />
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-179z3gj-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
          htmlFor=":r2:"
          id=":r2:-label"
        >
          Submit button
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary Mui-disabled MuiInputBase-formControl css-hjtp1-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            defaultValue="Outlined"
            disabled={true}
            id=":r2:"
            name="submit"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="submit"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-14lo706"
            >
              <span>
                Submit button
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </form>
</div>
`;
